src/query_plan/operation.rs

```diff
@@ -416,6 +418,28 @@ pub(crate) struct NormalizedFragment {
     pub(crate) selection_set: NormalizedSelectionSet,
 }
 
+impl NormalizedFragment {
+    fn normalize(
+        fragment: &Fragment,
+        schema: &ValidFederationSchema,
+    ) -> Result<Self, FederationError> {
+        Ok(Self {
+            schema: schema.clone(),
+            name: fragment.name.clone(),
+            type_condition_position: schema
+                .get_type(fragment.type_condition().clone())?
+                .try_into()?,
+            directives: Arc::new(fragment.directives.clone()),
+            selection_set: NormalizedSelectionSet::normalize_and_expand_fragments(
+                &fragment.selection_set,
+                &IndexMap::new(),
+                schema,
+                false,
+            )?,
+        })
```

Comment:
A downside of boolean arguments is that when a caller passes `true` or `false` there’s no indication of what that mean. Consider either:

* Replace `bool` with a dedicated two-variant enum
* Or introduce a named variable at such call sites. For example:

```suggestion
        let inline_fragment_spreads = false;
        Ok(Self {
            schema: schema.clone(),
            name: fragment.name.clone(),
            type_condition_position: schema
                .get_type(fragment.type_condition().clone())?
                .try_into()?,
            directives: Arc::new(fragment.directives.clone()),
            selection_set: NormalizedSelectionSet::normalize_and_expand_fragments(
                &fragment.selection_set,
                &IndexMap::new(),
                schema,
                inline_fragment_spreads,
            )?,
        })
```

Reply:
indeed, i like the explicit enum variant -> will update it